src/DynamoCoreWpf/Views/Search/LibrarySearchView.xaml.cs

```diff
@@ -0,0 +1,693 @@
+            }
+            if (HighlightedItem == null) return;
+
+            target = PresentationSource.FromVisual(HighlightedItem);
+            if (target == null)
+            {
+                // During search, backing data collections typically get updated frequently.
+                // This may result in corresponding NodeSearchElementViewModel being removed or 
+                // updated. When that happens, the visual element 'HighlightedItem' that gets 
+                // bound to the original NodeSearchElementViewModel then becomes DisconnectedItem.
+                // In such cases, we will reset the HighlightedItem to 'topResultListBox'.
+                HighlightedItem = GetSelectedListBoxItem(topResultListBox);
+                if (HighlightedItem == null) return;
+
+                target = PresentationSource.FromVisual(HighlightedItem);
+            }
+
+            var routedEvent = Keyboard.KeyDownEvent; // Event to send
+
+
+
+            HighlightedItem.RaiseEvent(new KeyEventArgs(Keyboard.PrimaryDevice,
+                                                                            target,
+                                                                            0,
+                                                                            key) { RoutedEvent = routedEvent });
+        }
+
+        private ListBoxItem GetSelectedListBoxItem(ListBox listbox)
+        {
+            if (!listbox.HasItems || (listbox.SelectedIndex == -1))
+                return null;
+
+            var generator = listbox.ItemContainerGenerator;
+            return generator.ContainerFromItem(listbox.SelectedItem) as ListBoxItem;
+        }
+
+        // This event is used to move inside members.
+        private void OnMembersListBoxKeyDown(object sender, KeyEventArgs e)
+        {
+            // For hovered by mouse item do not navigate.
+            if (HighlightedItem.IsMouseOver)
+            {
+                e.Handled = true;
+                return;
+            }
+
+            var selectedMember = HighlightedItem.DataContext as NodeSearchElementViewModel;
+            var membersListBox = sender as ListBox;
+            var members = membersListBox.Items;
+
+            int selectedMemberIndex = 0;
+            for (int i = 0; i < members.Count; i++)
+            {
+                var member = members[i] as NodeSearchElementViewModel;
+                if (member.Equals(selectedMember))
+                {
+                    selectedMemberIndex = i;
+                    break;
+                }
+            }
+
+            int nextselectedMemberIndex = selectedMemberIndex;
+            if (e.Key == Key.Down)
+                nextselectedMemberIndex++;
+            if (e.Key == Key.Up)
+                nextselectedMemberIndex--;
+
+            if (nextselectedMemberIndex < 0 || nextselectedMemberIndex > members.Count - 1)
+                return;
```

Comment:
For better readability, `if (nextselectedMemberIndex < 0 || nextselectedMemberIndex >= members.Count)`